SampleBrowser/SampleBrowser/UI/Suggestion/SuggestionViewController.swift

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+protocol SuggestionViewControllerDelegate: AnyObject {
+    func tapOnSuggestion(term: String)
+}
+
+class SuggestionViewController: UIViewController, UITableViewDelegate {
+    private var tableView: UITableView
+    private var dataSource: SuggestionDataSource!
+    private weak var delegate: SuggestionViewControllerDelegate?
+
+    private var gradientLayer: CAGradientLayer?
+    private var topIconConstraint: NSLayoutConstraint?
+
+    init() {
+        self.tableView = UITableView(frame: .zero, style: .grouped)
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Life cycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureTableView()
+
+        // Showing app logo when no search is visible
+        tableView.isHidden = true
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        setGradientBackground()
+        setTopConstraint()
+    }
+
+    private func setTopConstraint() {
+        let height = view.frame.height * 1 / 5
+        topIconConstraint?.constant = height
+    }
+
+    private func setGradientBackground() {
+        self.gradientLayer?.removeFromSuperlayer()
+        let colorTop =  UIColor.orange.cgColor
+        let colorBottom = UIColor.purple.cgColor
+
+        let gradientLayer = CAGradientLayer()
+        gradientLayer.colors = [colorTop, colorBottom]
+        gradientLayer.locations = [0.0, 0.6]
+        gradientLayer.frame = self.view.bounds
+        self.gradientLayer = gradientLayer
+        view.layer.insertSublayer(gradientLayer, at: 0)
+    }
+
+    private func configureTableView() {
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(tableView)
+
+        NSLayoutConstraint.activate([
+            tableView.topAnchor.constraint(equalTo: view.topAnchor),
+            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+
+        tableView.tableHeaderView = UIView(frame: CGRect(origin: .zero,
+                                                         size: CGSize(width: 0, height: CGFloat.leastNormalMagnitude)))
```

Comment:
What's the purpose of using `.leastNormalMagnitude` here? I'm guessing this is a workaround or UIKit hack of some kind?

Reply:
I think this is just to ensure there are no header view in the suggestion table view. 
If I remove that code I get this:
![Screenshot 2024-01-18 at 2 34 10 PM](https://github.com/mozilla-mobile/firefox-ios/assets/11338480/0248b60d-e4af-4352-b1fa-a313ed406160)

Instead of:
![Screenshot 2024-01-18 at 2 34 34 PM](https://github.com/mozilla-mobile/firefox-ios/assets/11338480/d5ea3692-58db-485e-a833-36c40bbea3c1)

There are no designs for this sample browser app, so it's only a visual choice we made in SwiftFox that was ported over here 🤔